c/parallel/include/cccl/c/for.h

```diff
@@ -11,13 +11,14 @@
 #pragma once
 
 #ifndef CCCL_C_EXPERIMENTAL
-#  warning "C exposure is experimental and subject to change. Define CCCL_C_EXPERIMENTAL to acknowledge this warning."
-#else // ^^^ !CCCL_C_EXPERIMENTAL ^^^ / vvv CCCL_C_EXPERIMENTAL vvv
+#  error "C exposure is experimental and subject to change. Define CCCL_C_EXPERIMENTAL to acknowledge this notice."
```

Comment:
As long as the team is OK with compile-time error, I am OK. Most likely this is not user-visible because we build/ship the binary.

Reply:
Other headers cause compile-time error if `CCCL_C_EXPERIMENTAL` is not set, so I went along. 

However, I feel these conditionals should be moved from header files to translation units that implement exported functionality. It is for a separate PR and requires a discussion.

Reply:
Error or warning is both fine, as long as you can suppres it somehow.